### ISSUE TITLE
Fix pagination on tables in the UI

### ISSUE THREAD

**Issue by lamppu** [eclipse-apoapsis/ort-server#989]
Seems like something has broken the pagination in the tables and the total page number is sometimes wrong and the buttons to the next/previous page aren't properly disabled anymore.

**Comment by Etsija** [comment]
PR #981, specifically [this commit](https://github.com/eclipse-apoapsis/ort-server/pull/981/commits/5686fc2adf302b4c780177e6de0ba7cb06cf05ca) was probably the one breaking pagination. I'll explain this using the  route `organizations/$orgId/index.tsx`:

1. Before the commit, we explicitly set the query keys for queries, and in this case it was
```
queryKey: [
  useProductsServiceGetOrganizationProductsKey,
  params.orgId,
  pageIndex,
  pageSize,
],
```

2. In PR #981, we switched to use pre-generated custom hooks for the queries, which should consolidate the query keys and functions together for us, to simplify queries. In this case, we use the `useProductsServiceGetOrganizationProducts` hook, which defines the query key using a special query key function `UseProductsServiceGetOrganizationProductsKeyFn`:
```
queryKey: Common.UseProductsServiceGetOrganizationProductsKeyFn(
  { limit, offset, organizationId, sort },
  queryKey
),
```

3. This function returns the key in form
```
[
  useProductsServiceGetOrganizationProductsKey,
  ...(queryKey ?? [{ limit, offset, organizationId, sort }]),
]

```
which means that if we don't specify our "own" query key, which should be a combination given in 1., the query key it returns is a combination of `[useProductsServiceGetOrganizationProductsKey, organizationId, limit, offset, sort]` instead of the correct one `[useProductsServiceGetOrganizationProductsKey, organizationId, pageIndex, pageSize]`. This leads to erroneous cache handling for the queries, in effect breaking pagination.

4. There might be an easier way to fix this, but at least this works: instead of calling the hooks "as-is"
```
const {
  data: products,
  error: prodError,
  isPending: prodIsPending,
  isError: prodIsError,
} = useProductsServiceGetOrganizationProducts(
  {
    organizationId: Number.parseInt(params.orgId),
    limit: pageSize,
    offset: pageIndex * pageSize,
  },
);
```
we provide the correct query keys explicitly to the hook with
```
const {
  data: products,
  error: prodError,
  isPending: prodIsPending,
  isError: prodIsError,
} = useProductsServiceGetOrganizationProducts(
  {
    organizationId: Number.parseInt(params.orgId),
    limit: pageSize,
    offset: pageIndex * pageSize,
  },
  [
    useProductsServiceGetOrganizationProductsKey,
    params.orgId,
    pageIndex,
    pageSize,
  ]
);
```

**Comment by Etsija** [comment]
It seems like in my PR #981 I accidentally left [this file](https://github.com/eclipse-apoapsis/ort-server/blob/587340b1f1fca9b149a306c1e54aa2e4c33e133e/ui/src/routes/_layout/organizations/%24orgId/products/%24productId/index.tsx#L99-L123) unchanged, which is handy, because that is the only page where pagination is correct.  This isolates the problem to PR #981, which certainly broke pagination.  I've been investigating the problem, and to me there seems to be something fishy going on in the query keys generated with the hooks we now use, but I haven't been yet to pinpoint the problem.

**Comment by mmurto** [comment]
@lamppu or @Etsija, can you check if the pagination is still broken with #999?

**Comment by Etsija** [comment]
I noticed the upcoming TypeScript 5.6 (on the way to be merged for our project) would probably have already [caught my mistake](https://devblogs.microsoft.com/typescript/announcing-typescript-5-6/#disallowed-nullish-and-truthy-checks1) 😃 .

**Comment by Etsija** [comment]
Closed with #1002.
